hydrolib/core/basemodel.py

```diff
@@ -38,22 +75,45 @@ class FileModel(BaseModel, ABC):
 
     filepath: Optional[Path] = None
 
-    def __init__(self, filepath: Optional[Path] = None, *args, **kwargs) -> None:
+    def __init__(self, filepath: Optional[Path] = None, *args, **kwargs):
         """Initialize a model.
 
         The model is empty (with defaults) if no `filepath` is given,
         otherwise the file at `filepath` will be parsed."""
         # Parse the file if path is given
+        token = None
         if filepath:
-            data = self._load(Path(filepath))  # so we also accept strings
+            filepath = Path(filepath)  # so we also accept strings
+
+            # If a context is set, use it
+            folder = context_dir.get(None)
+            if folder and not filepath.is_absolute():
+                logger.info(f"Used context to get {folder} for {filepath}")
+                filepath = folder / filepath
+            # Otherwise we're the root filepath
+            # and should set the context
+            else:
+                logger.info(f"Set context to {filepath.parent}")
+                token = context_dir.set(filepath.parent)
```

Comment:
I'm not entirely sure about this. It would make sense for a model to refer to the context of its parent for loading the file, but I vaguely recall something that relative paths of FMModels are always relative to the .mdu, even if they parent is located somewhere else.

Might be good to verify this with @arthurvd .

Reply:
Agreed, that's why we check for `is_absolute` above.